rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt

```diff
@@ -854,9 +854,9 @@ data class CheckStmt(
 
 @Serializable
 data class CheckrStmt(
-    val comparatorString: Expression, // Factor1
-    val baseString: Expression,
-    val start: Int = 1,
+    var comparatorString: Expression, // Factor1
```

Comment:
I think this property, along with the ones below, should remain immutable as they were before.

Reply:
This is needed to prevent mutability errors when expression is a function call or an array access expression. Here is an example of the very same issue fixed previously. there are many more identical to this in PR history: https://github.com/smeup/jariko/pull/595

Reply:
@lanarimarco while fixing for `CHECK` statement, I noticed that there is a `FactorContext.toIndexedExpression()` method that seems to do exactly what the new method `FactorContext.toPositionalExpression()` does with only support for literals. Currently it is only used for `SCAN` and `XLATE` statements. Do you think it is safe to state that `FactorContext.toIndexedExpression()` tries to do what `FactorContext.toPositionalExpression()` is now doing?

In that case I think it would be wise to drop the old method and replace it with the new one while keeping the old name. This should provide more clarity and avoid having duplicate symbols with different logic whose use is ambiguous.

Reply:
> This is needed to prevent mutability errors when expression is a function call or an array access expression. Here is an example of the very same issue fixed previously. there are many more identical to this in PR history: #595

Ah, okay, I remember now.  
Can you add a test? This will make the reason for your change clearer.  

Thank you.

Reply:
> In that case I think it would be wise to drop the old method and replace it with the new one while keeping the old name. This should provide more clarity and avoid having duplicate symbols with different logic whose use is ambiguous.

Absolutely agree.